src/GenericFactory.sol

```diff
@@ -153,4 +157,24 @@ contract GenericFactory is IGenericFactory, Owned {
     {
         return Address.functionCallWithValue(aTarget, aCalldata, aValue, "FACTORY: RAW_CALL_REVERTED");
     }
+
+    event Deployed(address aAddress, bytes32 aSalt);
+
+    /// @notice Deploys a given contract and salt using CREATE2.
+    /// @param  aInitCode   The contract init code to deploy.
+    /// @param  aSalt       A user-provided value that can be used to produce
+    ///                     varying addresses for the same init code.
+    /// @return rContract The address of the newly deployed contract.
+    function deploy(bytes memory aInitCode, bytes32 aSalt) external payable returns (address rContract) {
+        require(_deployInProgress, "AUTH: ONLY CHILDREN CAN CALL");
```

Comment:
should we change the error msg to `FACTORY: ONLY CHILDREN CAN CALL` to be more in line with the existing style?

Reply:
Yes, will do